proxy/src/control/destination/mod.rs

```diff
@@ -26,13 +30,20 @@ pub struct Resolver {
 #[derive(Debug)]
 struct ResolveRequest {
     authority: DnsNameAndPort,
+    respond: Respond,
+}
+
+#[derive(Debug)]
+struct Respond {
     update_tx: mpsc::UnboundedSender<Update>,
+    receiver_gone: Arc<AtomicBool>,
```

Comment:
This would be clearer if we used `Arc<()>` and `Weak<()>` respectively. Note in particular that there are two indications that the receiver is gone: There are no references to the the flag other than the Destination service's reference, and the value is true. By using `Weak<()>` for the destination service's reference we reduce this to one indicator.